tests/ui/question_mark.rs

```diff
@@ -583,3 +584,17 @@ fn issue_15679() -> Result<i32, String> {
 
     Ok(0)
 }
+
+mod issue14894 {
+    struct CannotCopy {}
+
+    fn use_after_question_mark(do_something_else: impl FnOnce() -> Result<CannotCopy, ()>) -> Result<(), ()> {
+        let result = do_something_else();
+        if let Err(reason) = result {
+            return Err(reason);
+        }
+        drop(result);
```

Comment:
Can you also add the following test?

```rust
        let result = do_something_else();
        let x = match result {
            //~^ question_mark
            Ok(v) => v,
            Err(e) => return Err(e),
        };
        drop(x);
```

(you will have to change `FnOnce` to `Fn` and `#![expect(clippy::drop_non_drop)]`)